deps/csl.mk

```diff
@@ -22,31 +22,28 @@ endef
 # would get from CSL (by searching for a `GLIBCXX_3.4.X` symbol that does not exist
 # in our CSL, but would in a newer one), and default to `USE_BINARYBUILDER_CSL=0` in
 # this case.
-CSL_NEXT_GLIBCXX_VERSION=GLIBCXX_3\.4\.30|GLIBCXX_3\.5\.|GLIBCXX_4\.
 
-# First, check to see if BB is disabled on a global setting
+# First, check to see if BB is disabled on a global setting.
+# If it's not, check to see if it's disabled by a USE_SYSTEM_xxx flag.
+# If it's not, see if we should disable it due to the system `libstdc++` being newer.
+# Otherwise, use binary builder, and use our bundled libstdc++.
 ifeq ($(USE_BINARYBUILDER),0)
 USE_BINARYBUILDER_CSL ?= 0
 else
-# If it's not, check to see if it's disabled by a USE_SYSTEM_xxx flag
 ifeq ($(USE_SYSTEM_CSL),1)
 USE_BINARYBUILDER_CSL ?= 0
 else
-# If it's not, see if we should disable it due to `libstdc++` being newer:
 LIBSTDCXX_PATH := $(eval $(call pathsearch,libstdc++,$(STD_LIB_PATH)))
 ifneq (,$(and $(LIBSTDCXX_PATH),$(shell objdump -p $(LIBSTDCXX_PATH) | grep $(CSL_NEXT_GLIBCXX_VERSION))))
-# Found `libstdc++`, grepped it for strings and found a `GLIBCXX` symbol
-# that is newer that whatever we have in CSL.  Default to not using BB.
-USE_BINARYBUILDER_CSL ?= 0
+USE_BINARYBUILDER_CSL ?= 0 # Found a newer symbol, default to not using BB
```

Comment:
Do we still need to do this?  @giordano I'm thinking we can actually drop this part (essentially reverting https://github.com/JuliaLang/julia/commit/27c0291f7fbecd811a3f9eb62b1159cff6b61e2a) and just always ship our `libstdc++` (as long as `USE_BINARYBUILDER_CSL=1`), since the decision to use it or not should now be made correctly at runtime.

Reply:
Uhm, yeah, good point, so we'd have a single place where we do this check (runtime), which would simplify things a bit.

Reply:
I've made the change because I agree. But, note that this does affect whether other libraries are copied through BB or not in that case. I take it that this is acceptable?

Reply:
Yes; we're just always going to copy the libraries through now.